### PR TITLE
MTM-43145 added a tutorial for testing

### PR DESCRIPTION
**Pull request by janhommes**

### CODE REVIEW

**Comment by janhommes** [comment]
@rssag @MWindel can we merge this?

File: content/web/tutorials-bundle/add-a-unit-test.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+---
+title: Add a Jest based unit test
```
Comment (BeateRixen) [inline content/web/tutorials-bundle/add-a-unit-test.md:2]: @rssag Jest based > Jest-based

File: content/web/tutorials-bundle/add-a-unit-test.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+However, any application supports unit tests in the same way. Next, you need to install all dependencies.
+
+Switch to the new folder and run `npm install`.
+
+>**Info:** The `c8ycli new` command has a `-a` flag which defines which package to use for scaffolding. This way you can also define which version of the application you want to scaffold, for example:
+>
+> - `c8ycli new my-cockpit cockpit -a @c8y/apps@1013.0.62` will scaffold an application with the version `1013.0.62`
+> - `c8ycli new my-cockpit cockpit -a @c8y/apps@latest` will scaffold an application with the latest official release. Same as if used without the `-a` flag
```
Comment (BeateRixen) [inline content/web/tutorials-bundle/add-a-unit-test.md:28]: @rssag There should be a full stop in all cases or never

File: content/web/tutorials-bundle/add-a-unit-test.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+Test Suites: 1 passed, 1 total
+Tests:       1 passed, 1 total
+Snapshots:   0 total
+Time:        32.858 s
+```
+
+If the test says `PASS`, everything went well and your first component test was successful.
+Now, you can add more detailed test cases to verify your component works as intended.
+
+### 3. Use a Snapshot Test to verify the component template
```
Comment (BeateRixen) [inline content/web/tutorials-bundle/add-a-unit-test.md:126]: @rssag Snapshot Test > snapshot test

File: content/web/tutorials-bundle/add-a-unit-test.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+Tests:       2 passed, 2 total
+Snapshots:   1 written, 1 total
+Time:        5.154 s
+```
+
+You can find and verify this snapshot in the newly created folder `./__snapshot__`.
+When the template changes, the test will fail and you need to overwrite your test with `npm test -- -u`.
+You can test this behavior by changing your template in the `test.component.ts` file.
+
+>**Info:** It is common practice to commit those snapshots with your code.
```
Comment (BeateRixen) [inline content/web/tutorials-bundle/add-a-unit-test.md:164]: those > these?

File: content/web/tutorials-bundle/add-a-unit-test.md (hunk 1 above)
Comment (rssag) [reply content/web/tutorials-bundle/add-a-unit-test.md:2]: done

File: content/web/tutorials-bundle/add-a-unit-test.md (hunk 3 above)
Comment (rssag) [reply content/web/tutorials-bundle/add-a-unit-test.md:126]: done

**Review by BeateRixen — APPROVED** [review]

**Comment by BeateRixen** [comment]
> @rssag @MWindel can we merge this?

Can be merged now. @janhommes Should that also go into 10.13?

**Comment by janhommes** [comment]
@BeateRixen Yes, 10.13 would be good. What do i need to do?

**Comment by BeateRixen** [comment]
I will cherry-pick that to 10.13.
________________________________

**Pull request merged**